lib/Dialect/Synth/SynthOps.cpp

```diff
@@ -67,12 +67,60 @@ llvm::APInt MajorityInverterOp::evaluate(ArrayRef<APInt> inputs) {
 OpFoldResult MajorityInverterOp::fold(FoldAdaptor adaptor) {
   // TODO: Implement maj(x, 1, 1) = 1, maj(x, 0, 0) = 0
 
+  // x x 1 -> x
+  // x 1 1 -> 1
+  // x ~x 1 -> 1
+  // x 0 0 -> 0
+  // x 1 1 -> 1
+
+  bool isOpConstant = true;
+  // for all constant inputs
   SmallVector<APInt, 3> inputValues;
+  size_t i = 0, index;
```

Comment:
Could you do slightly change the code structure to store non-constant index and check based on the number constants? 
```suggestion
  SmallVector<APInt, 3> inputValues;
  SmallVector<unsigned, 3> nonConstantIndex;
  
 for (auto [i, input] : llvm::enumerate(adaptor.getInputs())) {
    auto attr = llvm::dyn_cast_or_null<IntegerAttr>(input);
    if (attr) 
        inputValues.push_back(attr);
     else 
       nonConstantIndex.push_back(i);
  }
  
  if (inputValues.size() == getNumINputs())
    return IntegerAttr::get(getType(), evaluate(inputValues));  
    
  if (getNumOperands() != 3) 
    return {};
  
 if (nonConstantIndex.size() == 1) {
    // Get two constants. 
    auto idx = nonConstantIndex.front();
    auto constIndex1 = (idx+1)%3;
    auto constIndex2 = (idx+2)%3;
    auto c1 = adaptor.getInputs()[constIndex1];
    auto c2 = ...
    auto ...;
 }
 
 
```